src/OpenTelemetry.Instrumentation.AspNet/MeterProviderBuilderExtensions.cs

```diff
@@ -34,7 +34,9 @@ public static class MeterProviderBuilderExtensions
     {
         Guard.ThrowIfNull(builder);
 
+#pragma warning disable CA2000 // Dispose objects before losing scope
         var instrumentation = new AspNetMetrics();
+#pragma warning restore CA2000 // Dispose objects before losing scope
```

Comment:
Analysis might be happy (don't need the suppression) if you move this "new" into the factory like...

return builder.AddInstrumentation(() => new AspNetMetrics());

90% sure about that didn't try it to verify 😄